### ISSUE TITLE
Implement Baseline Prediction Model

### ISSUE THREAD

**Issue by Kame4201** [Kame4201/beat-books-model#2]
@claude 
**Labels:** `data-science`, `priority:high`, `phase:2-model`

### Summary

Build ML models for two prediction targets: Win/Loss (classification) and Point Spread (regression). Start with simple baselines, progress to gradient-boosted models.

**Depends on:** Feature Engineering Pipeline (Issue 1 in this repo)

### Tasks

**Create module structure:**
- [ ] `src/models/base_predictor.py` — abstract base class:
  ```python
  class BasePredictor(ABC):
      def train(self, X_train, y_train): ...
      def predict(self, X): ...
      def predict_proba(self, X): ...
      def evaluate(self, X_test, y_test) -> dict: ...
      def save(self, path: str): ...
      def load(cls, path: str): ...
  ```
- [ ] `src/models/model_config.py` — hyperparameter configurations
- [ ] `src/models/model_registry.py` — track trained model versions

**Win/Loss Classification:**
- [ ] `src/models/win_loss_model.py`
- [ ] Baseline: Logistic Regression (scikit-learn)
- [ ] Advanced: Gradient Boosted Classifier (XGBoost or LightGBM)
- [ ] Output: `P(home_win)`, `P(away_win)` summing to 1.0

**Spread Prediction:**
- [ ] `src/models/spread_model.py`
- [ ] Baseline: Linear Regression
- [ ] Advanced: Gradient Boosted Regressor
- [ ] Output: predicted point spread (negative = home favored)

**Train/test split — walk-forward by season (NOT random):**
- [ ] Train on 2000–2022, test on 2023–2024

**Model Registry:**
- [ ] Store per model: model_id, model_type, version, train_date, hyperparameters, feature_version, train_seasons, test_seasons, metrics
- [ ] Save as JSON metadata + joblib model artifact in `model_artifacts/`
- [ ] `model_artifacts/` is gitignored (large binary files)

**Training pipeline:**
- [ ] Load features from feature store
- [ ] Split by season (walk-forward)
- [ ] Train with config
- [ ] Evaluate on held-out test set
- [ ] Save artifact + metadata to registry
- [ ] Print evaluation summary

### Acceptance Criteria

- Win/Loss model achieves >55% accuracy on 2023–2024 test data
- Spread model MAE < 14 points
- Both output probability/confidence alongside predictions
- Model registry tracks version, date, metrics, feature_version
- Models save and load without data loss
- No data leakage (training never includes future games)
- Unit tests in `tests/test_models/` for train/predict/evaluate cycle

---Create Pull request
